corehq/toggles.py

```diff
@@ -170,10 +170,14 @@ def wrapped_view(request, *args, **kwargs):
                 if request.user.is_superuser:
                     from corehq.apps.toggle_ui.views import ToggleEditView
                     toggle_url = reverse(ToggleEditView.urlname, args=[self.slug])
-                    messages.warning(request, mark_safe((
-                        'This <a href="{}">feature flag</a> should be enabled '
-                        'to access this URL'
-                    ).format(toggle_url)))
+                    messages.warning(
+                        request,
+                        mark_safe((
+                            'This <a href="{}">feature flag</a> should be enabled '
+                            'to access this URL'
+                        ).format(toggle_url)),
+                        fail_silently=True,  # workaround for tests: https://code.djangoproject.com/ticket/17971
```

Comment:
I imagine this means we dont use the messages in any tests? And have no plans to?

Reply:
It is impossible to use messages in tests without mocking them or otherwise working around them. So sounds right. Note that this change wasn't actually required, though it leads to less opaque error messages than [the one here](https://travis-ci.org/dimagi/commcare-hq/jobs/585665460).
```
======================================================================
ERROR: custom.icds_reports.tests.test_views_first_days_of_month:TestAWCsCoveredView.test_chart_first_day_of_month
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/custom/icds_reports/tests/test_views_first_days_of_month.py", line 72, in test_chart_first_day_of_month
    response_june = self.view(request, step='chart', domain='icds-test')
  File "/vendor/lib/python3.6/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/vendor/lib/python3.6/site-packages/django/utils/decorators.py", line 67, in _wrapper
    return bound_func(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/toggles.py", line 176, in wrapped_view
    ).format(toggle_url)))
  File "/vendor/lib/python3.6/site-packages/django/contrib/messages/api.py", line 99, in warning
    fail_silently=fail_silently)
  File "/vendor/lib/python3.6/site-packages/django/contrib/messages/api.py", line 30, in add_message
    'You cannot add messages without installing '
django.contrib.messages.api.MessageFailure: You cannot add messages without installing django.contrib.messages.middleware.MessageMiddleware```

Reply:
I think it makes sense on this particular instance because any test that exercises toggle-based permission checks would run into this error otherwise. And this message is not mission-critical to show to users if for whatever other strange reason it actually failed outside a test.